### PR TITLE
Using ES6 syntax and reorganizing backend dir

### PR DESCRIPTION
**Pull request by TaraTritt**
- Updated backend to use the latest babel packages
- Using ES6 import/export syntax 
- Reorganized directory structure to follow more closely https://medium.com/codebase/structure-of-a-nodejs-api-project-cdecb46ef3f8

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -18,23 +19,15 @@
...
-    "babel-cli": "^6.26.0",
-    "babel-eslint": "^8.2.6",
-    "babel-plugin-transform-async-to-generator": "^6.24.1",
-    "babel-preset-env": "^1.7.0",
-    "babel-register": "^6.26.0",
+    "@babel/cli": "^7.1.2",
+    "@babel/core": "^7.1.2",
+    "@babel/node": "^7.0.0",
+    "@babel/preset-env": "^7.1.0",
+    "babel-eslint": "^10.0.1",
```
Comment (c0d3ster) [inline backend/package.json]: Was this also changed to @babel/ or does it remain under the babel- namespace?

File: backend/src/services/demux/updaters/blog/index.js [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+import deletePost from './deletepost'
+import editPost from './editpost'
+import likePost from './likepost'
 
 const account = process.env.EOSIO_CONTRACT_ACCOUNT
 
-module.exports = [
+export default [
   {
     actionType: `${account}::createpost`, // account::action name
```
Comment (c0d3ster) [inline backend/src/services/demux/updaters/blog/index.js]: Should this action name be camel case?

File: backend/src/services/demux/ActionHandler.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
-const { AbstractActionHandler } = require('demux')
-const mongoose = require('mongoose')
-const Post = require('../api/post/post.model')
-const BlockIndexState = require('../api/block-index-state/block-index-state.model')
-const io = require('../utils/io')
+import { AbstractActionHandler } from 'demux'
+import mongoose from 'mongoose'
+import { Post, BlockIndexState } from '../../models'
```
Comment (c0d3ster) [inline backend/src/services/demux/ActionHandler.js]: Is NODE_PATH supported on the backend?  That allows us to keep directory references absolute and would make more terse

File: backend/package.json (hunk 1 above)
Comment (TaraTritt) [reply backend/package.json]: No, actually looked this up myself - it has not been moved under the babel namespace

File: backend/src/services/demux/ActionHandler.js (hunk 3 above)
Comment (TaraTritt) [reply backend/src/services/demux/ActionHandler.js]: I meant to do this, but i forgot, thanks for the reminder. I'll look it up now to see if it's possible

File: backend/src/services/demux/updaters/blog/index.js (hunk 2 above)
Comment (TaraTritt) [reply backend/src/services/demux/updaters/blog/index.js]: When i initial wrote the contract code i followed the action naming conventions in a few sample contracts that i had found, but it looks like snake case is pretty popular. I may eventually switch to that, but definitely think that's outside of the scope of this PR

File: backend/src/services/demux/ActionHandler.js (hunk 3 above)
Comment (TaraTritt) [reply backend/src/services/demux/ActionHandler.js]: Looks like setting NODE_PATH in .env does not work and using NODE_PATH is not a recommended practice for node anyways

**Pull request merged**